### DISCUSSION TITLE
Hardware requirements

### DISCUSSION THREAD

**Discussion by primski** [blakeblackshear/frigate#4152] [General]
Hey,

I am using a HP EliteDesk mini pc ... it has i7 6th gen CPU with 32 GB ram ... GPU integrated i believe

not a great thing for this by any means but not too shabby either if you ask me. 

but frigate is absolutely killing it; the fans will blow off and the thing will melt in a couple of months if i leave it on ...

this is probably expected? or is it ffmpeg killing it ? can it be optimized? 
thanks

**Comment by blakeblackshear** [comment]
It should be able to handle it fine. My guess is that you are trying to run detect on a high resolution/fps stream.

**Reply by primski** [reply]
Hmm, i run at the resolution the camera provides and the fps is 5 or 6, which isn't great by any standards

but the stream resolution is 3840x2160

should i try to lower it to 720p? - this will also use CPU though, right? 

many thanks!

**Reply by NickM-27** [reply]
Are you sure your camera doesn't have sub streams?  Running at `3840x2160` is no joke at all and would certainly explain high CPU usage

**Reply by primski** [reply]
hm i have no idea ... how would i go about figuring that out?

Here is the ffprobe output

```
frigate$ ffprobe version 4.2.7-0ubuntu0.1 Copyright (c) 2007-2022 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
  configuration: --prefix=/usr --extra-version=0ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, rtsp, from 'rtsp://user:pass@10.0.23.50:554/user=admin_password=C6FScWt3_channel=1_stream=0':
  Metadata:
    title           : RTSP Session
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: hevc (Main), yuvj420p(pc), 3840x2160 [SAR 4:3 DAR 64:27], 12 fps, 12 tbr, 90k tbn, 12 tbc
    Stream #0:1: Audio: pcm_alaw, 8000 Hz, 1 channels, s16, 64 kb/s
```

Interestingly enough, ffprobe command does not exit ... i had to CTRL-C it after a minute or so

I also have this in global settings ( not under cameras:)


```
ffmpeg:
  hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p

```
and this under camera definition
```

      output_args:
        record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac

```

**Reply by NickM-27** [reply]
What camera do you have?

**Reply by primski** [reply]
let me just add, the server is also running some other services, it'd not dedicated ... but only for home usage so not many users, in fact just one. and it's dockerized, so probably doesn't help with performance but not sure it slows down that much.

**Reply by primski** [reply]
not sure if product link are allowed here: https://www.aliexpress.com/item/1005004065199917.html?spm=a2g0o.order_list.0.0.21ef1802KSL7ii

it's some kind of Hikvision clone.

**Reply by NickM-27** [reply]
Hikvision have substreams so it definitely would I'd think, does it have a web interface or an app that can configure it?

**Reply by primski** [reply]
app yes, web interface is buggy does not work, but it does exist ...

it should also support ONVIF but every onvif device manager i tried, works only partially. won't connect, etc, etc until camera is rebooted.

ok many thanks, i will look into this; i don't even know what a substream is tbh :facepalm

**Reply by NickM-27** [reply]
a substream is just a stream the camera puts out that is lower than its native resolution. This is why the docs recommend name brand cameras with interfaces, because they come with the ability to customize the substreams so you can have high resolution for recordings and have the camera output a smaller stream for running detection on

**Reply by primski** [reply]
aha ok i see, Android app has no such options, maybe the web interface does ... i will go recheck this. it requires ActiveX enabled Internet explorer and haven't used that in ages :D 

so just to be clear .. this kind of set up would enable detection on lower resolution stream while recordings would still be high res? this sounds pretty cool :) thanks a lot!

**Reply by NickM-27** [reply]
yes, read https://docs.frigate.video/guides/camera_setup#example-camera-configuration for more details

**Reply by primski** [reply]
hm, maybe it does have ...

running ffprobe with ?stream=1 shows different output
` Stream #0:0: Video: hevc (Main), yuvj420p(pc), 800x448 [SAR 4:3 DAR 50:21], 12 fps, 12 tbr, 90k tbn, 12 tbc`

i will try and see what happens, many thanks most helpful

**Reply by NickM-27** [reply]
Yes, that would be a substream. https://docs.frigate.video/guides/getting_started#step-7-enable-recording-optional shows how to setup the main stream for recording and the sub stream for detect

**Reply by primski** [reply]
ok cool, i set it up. seems to be working. i had to re-set the mask coordinates, makes sense.
will leave it on for a few days, see what happens :) many thanks

**Comment by NickM-27** [comment]
Recommend reading https://github.com/blakeblackshear/frigate/issues/3860 even if you don't have a coral

**Reply by primski** [reply]
hm yea, i tried some of these already .. i will go though it again, thanks
